src/qt/walletview.cpp

```diff
@@ -333,9 +333,11 @@ void WalletView::unlockWallet(bool fForMixingOnly)
 {
     // Unlock wallet when requested by wallet model
     if (walletModel->getEncryptionStatus() == WalletModel::Locked || walletModel->getEncryptionStatus() == WalletModel::UnlockedForMixingOnly) {
-        auto dlg = new AskPassphraseDialog(AskPassphraseDialog::Unlock, this);
-        dlg->setModel(walletModel);
-        GUIUtil::ShowModalDialogAndDeleteOnClose(dlg);
+        AskPassphraseDialog dlg(AskPassphraseDialog::Unlock, this);
```

Comment:
It seems as gui#336 introduced one extra bug: [there's](https://github.com/dashpay/dash/pull/6569/commits/3a5f35eb052a342687ae5bc13828e703d0eb146f#diff-e77cbb2f5167a9c9fd417cfbc346698aa9424ffa7ff48506e7132a50f9eb2201L338-R336) missing usage of flag `fForMixingOnly`

This line should be:
```
AskPassphraseDialog dlg(fForMixingOnly ? AskPassphraseDialog::UnlockMixing : AskPassphraseDialog::Unlock, this);
```

Reply:
Oh, wow. Good catch! 👍